cpu/kinetis_common/hwtimer_arch.c

```diff
@@ -148,51 +251,42 @@ void hwtimer_arch_disable_interrupt(void)
 void hwtimer_arch_set(unsigned long offset, short timer)
 {
     (void)timer; /* we only support one timer */
-    stimer.counter32b += lptmr_get_cnr();
-    hwtimer_stop();
-
-    stimer.cmr32b = stimer.counter32b + offset;
+    unsigned mask = disableIRQ();
+    HWTIMER_STATS_INC(count_set_total);
+    /* it is important that we don't get interrupted between stopping and
+     * starting the timer again in order to not increase the risk of dropping
+     * any ticks */
     stimer.diff = offset;
 
     if (stimer.diff > LPTMR_MAXTICKS) {
         stimer.diff = LPTMR_MAXTICKS;
     }
 
-    DEBUG("cntr: %lu, cmr: %lu, diff: %lu\n", stimer.counter32b, stimer.cmr32b, stimer.diff);
+    uint32_t diff = stimer.diff;
+    uint32_t cnr = lptmr_update_cmr(diff);
+    ++stimer.counter32b;
```

Comment:
Can you please change it to stimer.counter32b += 1; , just for readability and move the comment from L143:L144 here?